### PR TITLE
feat (docs): Migration Guide

### PR DESCRIPTION
**Pull request by mrivasperez**
In this PR, I added my notes from the FS11 to FS12 migration into a new `Migration` page within the `Usage` section of docs.

### CODE REVIEW

File: docs/src/lib/config.ts [hunk 1]
```diff
@@ -41,7 +41,7 @@ export const SIDEBAR = {
   ],
   Usage: [
     { text: "Integration", link: "/en/usage/integration" },
-    { text: "Contribution", link: "/en/usage/contribution" },
+    { text: "Contribution", link: "/en/usage/contribution" }, { text: "Migration", link: "/en/usage/migration" },
```
Comment (crherman7) [inline docs/src/lib/config.ts:44]: format

File: docs/src/pages/en/usage/migration.mdx [hunk 2]
```diff
@@ -0,0 +1,59 @@
+---
+title: Migration
```
Comment (crherman7) [inline docs/src/pages/en/usage/migration.mdx:2]: This isn't technically correct for the rest of the community. Remember this is open-source and if this does grab traction most people will not have migrated from `@brandingbrand/flagship`

File: docs/src/pages/en/usage/migration.mdx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+### Android
+#### White Text
+> *Note: This will be resolved in the next FS12 release (0.0.11)*
+After upgrading, all text elements in Android that do not have a value for their color will be white when the device is in dark mode. To resolve this, ASSOS patched the template style values XML file so that the application defaulted to light mode, regardless of the device’s current theme (*[See PR](https://github.com/brandingbrand/assos/pull/682)*).
+
+#### Java Vesion
+It's possible OS updates have caused this issue when reported; however, if a Java version issue is encountered while building the Android application, you can resolve it by making sure you set the JAVA_HOME path equal to the one used by Android Studio in your path file.
+
+Open and edit your .zshrc file `code .zshrc`
```
Comment (crherman7) [inline docs/src/pages/en/usage/migration.mdx:26]: not everybody use zsh

File: docs/src/pages/en/usage/migration.mdx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+- Open Android Studio
+- Click on the more actions drop-down in the Welcome Screen.
+- Select SDK Manager
+- View Appearance & Behavior > System Settings > Android SDK
+- Make sure Android SDK Command-line Tools (latest) is selected and installed
+- Click on Apply / Ok
+
+If you note that you already have Command-line Tools installed via Android Studio, you should confirm that your path file correctly lists necessary Android development tools.
+
+- Open and edit your .zshrc file `code .zshrc`
```
Comment (crherman7) [inline docs/src/pages/en/usage/migration.mdx:50]: not everybody uses zsh

File: docs/src/pages/en/usage/migration.mdx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+export JAVA_HOME="/Applications/Android Studio.app/Contents/jre/Contents/Home"
+```
+
+#### Command Line Tools
+If you receive an error message regarding command line tools, such as the following:
+```
+> Cannot run program "/Users/User/Library/Android/sdk/cmdline-tools/latest/bin/sdkmanager": error=2, No such file or directory
+```
+
+It’s likely you need to install the Android Command-line Tools within Android Studio.
```
Comment (crherman7) [inline docs/src/pages/en/usage/migration.mdx:39]: not sure I agree with including all of this - this all comes straight out of the react native environment setup documentation

File: docs/src/pages/en/usage/migration.mdx [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+- View Appearance & Behavior > System Settings > Android SDK
+- Make sure Android SDK Command-line Tools (latest) is selected and installed
+- Click on Apply / Ok
+
+If you note that you already have Command-line Tools installed via Android Studio, you should confirm that your path file correctly lists necessary Android development tools.
+
+- Open and edit your .zshrc file `code .zshrc`
+- Make sure the file contains the following lines for Android Studio.
+
+```
```
Comment (crherman7) [inline docs/src/pages/en/usage/migration.mdx:53]: not sure I agree with including all of this - this all comes straight out of the react native environment setup documentation

File: docs/src/pages/en/usage/migration.mdx (hunk 2 above)
Comment (mrivasperez) [reply docs/src/pages/en/usage/migration.mdx:2]: Maybe we can rename it to `Troubleshooting`?

File: docs/src/pages/en/usage/migration.mdx (hunk 3 above)
Comment (mrivasperez) [reply docs/src/pages/en/usage/migration.mdx:26]: I'll add a note 👍🏽

File: docs/src/pages/en/usage/migration.mdx (hunk 6 above)
Comment (mrivasperez) [reply docs/src/pages/en/usage/migration.mdx:53]: I think renaming it to `Troubleshooting` and keeping all these details on there would be helpful for current and new users.

I didn't copy and paste this from the react-native documentation. I created these instructions while helping @scbb31994 and @BBryanJ troubleshoot issues we could not solve following the react-native development environment setup docs after an upgrade to FS12.

File: docs/src/lib/config.ts (hunk 1 above)
Comment (jasonmosley) [reply docs/src/lib/config.ts:44]: @mrivasperez add a return here to match the formatting and also update the naming and file naming as well. Overall I like this content and would like to get it merged in.

**Review by jasonmosley — APPROVED** [review]

**Pull request merged**